esphome/onju-voice-microwakeword.yaml

```diff
@@ -235,18 +238,16 @@ voice_assistant:
         green: 100%
         effect: speaking
   on_end:
-    - delay: 200ms
     - wait_until:
         not:
-          media_player.is_playing: onju_out
+          voice_assistant.is_running:
```

Comment:
Were it not for this change, i'd have approved it in a heartbeat. Since this concerns other optimizations, not just fixing the config compatibility, i'll have to test things out.

Reply:
Yes, I prefer if someone with more experience about the media_player statemachines would check this, as I have the feeling that this leads to a bit of an increased latency after the TTS response is given.

Reply:
Without checking, i am fairly sure that `voice_assistant.is_running` turns `false` as soon as the TTS URL is passed to the `media_player`, so the MWW listening starts before longer response finishes being played.

Reply:
[There seems to be a 2 second timeout](https://github.com/esphome/esphome/blob/0ef73c6dd6b6a02b51f0effa9f509cb9dc7189d7/esphome/components/voice_assistant/voice_assistant.cpp#L390), but yes you are right. Then my solution here is not good. I wonder why the `media_player.is_playing` often times does not change to false though.

Reply:
I have changed this back to `media_player.is_playing`. I think I was confused by another instability and the missing "DETECTING_WAKEWORD" state-change message (or I didn't see it). Seems to be working okay, except for the microphone queue sometimes running full and the device restarting. Not our bug though, and perhaps it was already happening before?

Reply:
I had another look at my logs, the order seems to be like this, surprisingly:

```
[13:24:09][I][esp_adf_pipeline:132]: [ i2s_in ] status: 12
[13:24:09][D][adf_audio_element:191]: [pcm_reader] Checking State, got 65
[13:24:09][D][esp_adf_pipeline:448]: [ADFMicrophone] Pipeline changed from STARTING to RUNNING. (REQ: 0)
[13:24:09][D][micro_wake_word:195]: State changed from STARTING_MICROPHONE to DETECTING_WAKE_WORD
[13:24:09][D][adf_audio_element:191]: [http] Checking State, got 79
[13:24:09][D][adf_audio_element:191]: [decoder] Checking State, got 79
[13:24:09][I][esp_audio_sources:033][http]: Receive http event: 2
[13:24:09][I][esp_audio_sources:033][http]: Receive http event: 4
[13:24:10][D][esp-idf:000][http]: I (88344) HTTP_CLIENT: Body received in fetch header state, 0x3fcc0b03, 1841

[13:24:10][D][esp-idf:000][http]: I (88348) HTTP_STREAM: total_bytes=18862

[13:24:10][I][HTTPStreamReader:230]: Codec Format reported: 3.
[13:24:10][I][HTTPStreamReader:240]: [ * ] Receive music info from decoder, sample_rates=24000, bits=16, ch=1
[13:24:10][I][HTTPStreamReader:243]: [ * ] Receive music info from decoder, codec_fmt=3, bps=48000, duration=0, bytes=-1150
[13:24:10][D][adf_i2s_out:141]: Set final i2s settings: 16000
[13:24:10][D][esp_audio_processors:124]: Current settings: SRC: rate: 24000, ch: 1 bits: 16, DST: rate: 16000, ch: 1, bits 16
[13:24:10][D][adf_audio_element:108]: Preparing [http]...
[13:24:10][D][adf_audio_element:108]: Preparing [decoder]...
```

So the `DETECTING_WAKE_WORD` state is reached before the actual media is played on the media_player. Maybe this is related to the announcements feature in `esphome_audio`? This is not caused by this PR AFAIK. I think this is fine, it does seem to work. But in the worst case, there might a false detection on the TTS response - didn't happen to me yet though :)